src/proc/interface.rs

```diff
@@ -1,34 +1,41 @@
 use crate::arena::{Arena, Handle};
 
-struct Interface<'a> {
+pub struct Interface<'a, T> {
     expressions: &'a Arena<crate::Expression>,
-    uses: Vec<crate::GlobalUse>,
+    visitor: &'a mut T,
 }
 
-impl<'a> Interface<'a> {
-    fn add_inputs(&mut self, handle: Handle<crate::Expression>) {
+pub trait Visitor {
+    fn visitor_expr(&mut self, _: &crate::Expression) {}
```

Comment:
this start with a verb, which is "visit", not "visitor". So this would be `visit_expr` and `visit_lhs_expr`